overlord/devicestate/handlers_install.go

```diff
@@ -128,20 +128,17 @@ func (m *DeviceManager) doSetupRunSystem(t *state.Task, _ *tomb.Tomb) error {
 	if err != nil {
 		return err
 	}
+	bopts.Encrypt = useEncryption
 
 	var trustedInstallObserver *boot.TrustedAssetsInstallObserver
 	// get a nice nil interface by default
 	var installObserver install.SystemInstallObserver
-	if useEncryption {
-		bopts.Encrypt = true
-
-		trustedInstallObserver, err = boot.TrustedAssetsInstallObserverForModel(deviceCtx.Model(), gadgetDir)
-		if err != nil && err != boot.ErrObserverNotApplicable {
-			return fmt.Errorf("cannot setup asset install observer: %v", err)
-		}
-		if err == nil {
-			installObserver = trustedInstallObserver
-		}
+	trustedInstallObserver, err = boot.TrustedAssetsInstallObserverForModel(deviceCtx.Model(), gadgetDir, useEncryption)
+	if err != nil && err != boot.ErrObserverNotApplicable {
+		return fmt.Errorf("cannot setup asset install observer: %v", err)
+	}
+	if err == nil {
+		installObserver = trustedInstallObserver
```

Comment:
my preference would be to set trustedInstallObserver to nil immediately here, if !useEncrpytion